neo/rawio/intanrawio.py

```diff
@@ -72,34 +76,104 @@ class IntanRawIO(BaseRawIO):
                 self.filename
             )
+        # 3 possibilities for rhd files, one combines the header and the data in the same file with suffix `rhd` while
+        # the other two separates the data from the header which is always called `info.rhd`
+        # attached to the actual binary file with data
         elif self.filename.endswith(".rhd"):
-            self._global_info, self._ordered_channels, data_dtype, header_size, self._block_size = read_rhd(
-                self.filename
+            if filename.name == "info.rhd":
+                # first we have one-file-per-signal which is where one neo stream/file is saved as .dat files
+                if any((filename.parent / file).exists() for file in one_file_per_signal_filenames):
+                    self.file_format = 'one-file-per-signal'
+                    raw_file_paths_dict = create_one_file_per_signal_dict(filename.parent)
+                # then there is one-file-per-channel where each channel in a neo stream is in its own .dat file
+                else:
+                    self.file_format = 'one-file-per-channel'
+                    raw_file_paths_dict = create_one_file_per_channel_dict(filename.parent)
+            # finally the format with the header-attached to the binary file as one giant file
+            else:
+                self.file_format = 'header-attached'
+
+            self._global_info, self._ordered_channels, data_dtype, header_size, self._block_size, channel_number_dict = read_rhd(
+                self.filename, self.file_format
             )
 
         # memmap raw data with the complicated structured dtype
-        self._raw_data = np.memmap(self.filename, dtype=data_dtype, mode="r", offset=header_size)
+        # if header-attached there is one giant memory-map
+        if self.file_format == "header-attached":
+            self._raw_data = np.memmap(self.filename, dtype=data_dtype, mode="r", offset=header_size)
+        else:
+            self._raw_data = {}
+            for stream_index, (stream_index_key, stream_datatype) in enumerate(data_dtype.items()):
+                # for 'one-file-per-signal' we have one memory map / neo stream
+                if self.file_format == "one-file-per-signal":
+                    n_chans = channel_number_dict[stream_index_key]
+                    if stream_index_key == 4 or stream_index_key == 5:
+                        n_samples = int(os.path.getsize(raw_file_paths_dict[stream_index_key]) / 2) # uint16 2 bytes
+                    else:
+                        n_samples = int(os.path.getsize(raw_file_paths_dict[stream_index_key]) / (n_chans * 2))# unit16 2 bytes
+                    if stream_index_key != 6:
+                        self._raw_data[stream_index] = np.memmap(
+                            raw_file_paths_dict[stream_index_key], dtype=stream_datatype, shape = (n_chans, n_samples),  mode="r"
+                        ).T
```

Comment:
@samuelgarcia,

neo is samples x channels  so I added the transpose. But only your final read through just double check me? The Intan documentation says that map should be made with n_chans x n_samples

Reply:
Oh, no, does this mean that we are supposed to wait for @samuelgarcia ? : P

Well, call me back in a couple of months and I will do the profile then : )

Reply:
Haha. After our extensive testings I think he can do a lighter read. We've worked out most of the kinks and tests are passing, so we are in a good place.